cogniac/edgeflow.py

```diff
@@ -259,7 +259,16 @@ def upgrade(self, software_version):
         args = dict()
         if software_version is not None:
             args['software_version'] = software_version
-        self._cc._post("/gateways/%s/event/upgrade" % (self.gateway_id), data=args)
+        self._cc._post("/gateways/%s/event/upgrade" % (self.gateway_id), json=args)
```

Comment:
Just a comment—it's probably better to set `json=` than `data=` for API requests in the SDK unless JSON is decidedly not the format being exchanged for a given endpoint. Using it sets the headers correctly.